src/hitch.c

```diff
@@ -2471,10 +2498,12 @@ handle_mgt_rd(struct ev_loop *loop, ev_io *w, int revents)
 		(worker_state == WORKER_EXITING) ? "EXITING" : "ACTIVE");
 	}
 	else if (wu.type == BACKEND_REFRESH) {
-		backaddr->ai_family = wu.payload.addr.family;
-		backaddr->ai_addrlen = wu.payload.addr.len;
-		memcpy(backaddr->ai_addr, &wu.payload.addr.addr,
-		       sizeof(struct sockaddr_storage));
+		if (backaddr != NULL)
```

Comment:
`free(NULL)` is totally legitimate in C, so this `if (backaddr != NULL)` test could just be killed